### PR TITLE
Add with NextAuth.js example

### PR DESCRIPTION
**Pull request by earthlingdavey**
Hi Rob, thanks for the useful repo. I've forked it and added code for 'with-next-auth'. Would you be interested in taking a look at the code and possibly merging it in?
Cheers, Davey

### CODE REVIEW

**Comment by earthlingdavey** [comment]
BTW. I also learned 2 nice tips relating to neo4j and docker here. So thank-you :D

1. Using a Dockerfile with APOC downloaded in it, is bettter than using the official neo4j Dockerfile with the env var `NEO4JLABS_PLUGINS=["apoc"]`. Now I don't need to comment/un-coment that line for fresh installs.
2. The line `/usr/src/app/node_modules` in `docker-compose.yaml`. I didn't know you could do this, I usually make a named volume, but this is shorter :)

**Comment by TheRobBrennan** [comment]
@EarthlingDavey This is so awesome!!!! I'm hella swamped in meetings today, but will definitely take a look over the next day or so. As a fun note, I also just recently upgraded to the new 14" MacBook Pro, so I was curious about how Docker and Neo4j might work - and it appears you nailed it.

The NextAuth addition is huge! Thank you for submitting this. I will most definitely sit down and spend some time with it first. Also loved the Docker discoveries you shared. I wasn't aware of those two great points myself!

**Comment by earthlingdavey** [comment]
Hehe, good to see such a positive reaction! I'm in no rush here btw.

Re: docker and M1, I had a ball of a time with this 🤣 I need it for local dev on my M1 mac mini with 8GB of ram. The experimental image worked well for light use but for importing 10s of CVSs of > about 50k rows I kept running out of memory. It would also slow all the other apps to a crawl.

My solution was to crack out an 8 y/o i7 laptop, max out the memory to 16GB for £30, network it and run neo4j on bare metal Ubuntu. The difference is noticble, and I get no more memory issues, and apps on my mac are back to being snappy.

File: neo4j/v4.x.x/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Official Neo4j Docker Images available at https://hub.docker.com/_/neo4j
 FROM neo4j:4.2.2
 
+# OPTIONAL: For arm64 (M1 Macs) comment the previous line, and uncomment the next line.
```
Comment (TheRobBrennan) [inline neo4j/v4.x.x/Dockerfile:4]: This is such a great addition! I'm getting this project setup on my new 14" MacBook before reviewing your PR, and will use this line as well.

Comment (earthlingdavey) [reply neo4j/v4.x.x/Dockerfile:4]: Happy to have helped. Also, I've found that most other docker imags don't use a different repo url for arm64 compatible builds.
Enjoy your new MacBook!

Comment (TheRobBrennan) [reply neo4j/v4.x.x/Dockerfile:4]: Yeah! I'm also seeing in the live announcement from Neo4j - happening right now at https://www.youtube.com/embed/19m4MBYBnCs?autoplay=1&feature=oembed&wmode=opaque to discuss the new 4.4 release - that the generic base Neo4j image should work on arm64 as of tomorrow. That's exciting! We'll have to see how that goes.

Comment (TheRobBrennan) [reply neo4j/v4.x.x/Dockerfile:4]: Also I am happy to see my workload is lightening up a little, and I can finally take a look at your PR by the end of the weekend. Hopefully sooner!

Comment (earthlingdavey) [reply neo4j/v4.x.x/Dockerfile:4]: Cheers bud :) Look forward to seeing what you think...
BTW I coded up the neo4j adapter, so you might be the about the 2nd person to spin it up and test it out 😄 
Have a good weekend.

Comment (TheRobBrennan) [reply neo4j/v4.x.x/Dockerfile:4]: Hey hey! Just coming up for air after being buried under a mountain of things. I didn't want you to think I forgot about you or this PR. It's definitely on my radar - albeit a couple days out at this point given the mountain of stuff I'm digging out from 🏔

Comment (earthlingdavey) [reply neo4j/v4.x.x/Dockerfile:4]: Hey! No sweat - seriously. I get how workloads can be huge, so please don't feel commited to take a look at this quickly. The last thing I want is to know I'm contributing towards stresses. Projects like this should be fun, so just take a look if and when you feel like! ✌️

**Comment by TheRobBrennan** [comment]
@EarthlingDavey Hm I'm working on reviewing the app setup, and I love how you updated the main `README` - super cool.

I'm running into an issue on my machine, though, when trying to spin up the app using the `dev:clean` script. Next.js is unhappy and errors out during the `Generating static pages` step:

Screen Shot 2021-12-11 at 01 15 53 PM

Investigating...but if you find a solution before I do, lemme know. I also realized I can't commit any changes to this PR branch (obv reasons) but I may make a new one if I find a way to resolve this.

**Comment by TheRobBrennan** [comment]
@EarthlingDavey Aha! I think I'm on to something here.

I'm going to close this PR and create a new one.

**Comment by TheRobBrennan** [comment]
@EarthlingDavey Okey doke. Your example - along with a few tweaks I introduced - is now in the `main` branch of this project.

Thank you for your contribution! What a great addition to have. Much love, brotha' ::fist bump::

**Comment by earthlingdavey** [comment]
It's been a hectic weekend, so I missed your question. Sorry, I forgot to try and build it, - I only ran the next dev script. Hope it wasn't too much of a PITA to fix. I'll read you PR as a learning exercise.
Thanks for your time and adding it to main, have a top week 👊✌️

**Comment by TheRobBrennan** [comment]
All good @EarthlingDavey - this was a great addition. Thank you!

**Pull request closed without merging**